src/Routing/RouterListener.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+namespace Cache\CacheBundle\Routing;
+
+use Cache\Taggable\TaggablePoolInterface;
+use Psr\Cache\CacheItemPoolInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class RouterListener
+{
+    /**
+     * @var CacheItemPoolInterface
+     */
+    private $cache;
+
+    /**
+     * @var int
+     */
+    private $ttl;
+
+    /**
+     * @param CacheItemPoolInterface $cache
+     * @param int                    $ttl
+     */
+    public function __construct(CacheItemPoolInterface $cache, $ttl)
+    {
+        $this->cache = $cache;
+        $this->ttl   = $ttl;
+    }
+
+    /**
+     * Before routing, try to fetch route result from cache.
+     *
+     * @param GetResponseEvent $event
+     */
+    public function onBeforeRouting(GetResponseEvent $event)
+    {
+        $request = $event->getRequest();
+
+        if ($request->attributes->has('_controller')) {
+            // routing is already done
+            return;
+        }
+
+        $item = $this->getCacheItem($request);
+        if (!$item->isHit()) {
+            return;
+        }
+
+        $request->attributes->add($item->get());
+        $request->attributes->set('_cache_hit', true);
+    }
+
+    /**
+     * After the routing, make sure to store the result in cache.
+     *
+     * @param GetResponseEvent $event
+     */
+    public function onAfterRouting(GetResponseEvent $event)
+    {
+        $request = $event->getRequest();
+
+        if (!$request->attributes->has('_controller')) {
+            // routing has not taken place
```

Comment:
You missed the "!". 

If NOT the $request has _controller, no routing has taken place